qiskit_nature/problems/second_quantization/vibrational/vibrational_problem.py

```diff
@@ -43,12 +55,119 @@ def second_q_ops(self) -> List[SecondQuantizedOp]:
-        vibrational_spin_op = build_vibrational_op(watson_hamiltonian_transformed,
+        self._molecule_data: WatsonHamiltonian = cast(WatsonHamiltonian, self.driver.run())
+        self._molecule_data_transformed: WatsonHamiltonian = \
+            cast(WatsonHamiltonian, self._transform(self._molecule_data))
+
+        vibrational_spin_op = build_vibrational_op(self._molecule_data_transformed,
                                                    self.num_modals,
                                                    self.truncation_order)
 
-        second_quantized_ops_list = [vibrational_spin_op]
+        num_modes = self._molecule_data_transformed.num_modes
+        if isinstance(self.num_modals, int):
+            num_modals = [self.num_modals] * num_modes
+        else:
+            num_modals = self.num_modals
+
+        second_quantized_ops_list = [vibrational_spin_op] + create_all_aux_operators(num_modals)
 
         return second_quantized_ops_list
+
+    def hopping_ops(self, qubit_converter: QubitConverter,
+                    excitations: Union[str, int, List[int],
+                                       Callable[[int, Tuple[int, int]],
+                                                List[Tuple[Tuple[int, ...], Tuple[int, ...]]]]
+                                       ] = 'sd',
+                    ) -> Tuple[Dict[str, PauliSumOp], Dict[str, List[bool]],
+                               Dict[str, Tuple[Tuple[int, ...], Tuple[int, ...]]]]:
+        """Generates the hopping operators and their commutativity information for the specified set
+        of excitations.
+
+        Args:
+            qubit_converter: the `QubitConverter` to use for mapping and symmetry reduction. The
+                             Z2 symmetries stored in this instance are the basis for the
+                             commutativity information returned by this method.
+            excitations: the types of excitations to consider. The simple cases for this input are:
+                - a `str` containing any of the following characters: `s`, `d`, `t` or `q`.
+                - a single, positive `int` denoting the excitation type (1 == `s`, etc.).
+                - a list of positive integers.
+                - and finally a callable which can be used to specify a custom list of excitations.
+                  For more details on how to write such a function refer to the default method,
+                  :meth:`generate_vibrational_excitations`.
+
+        Returns:
+            A tuple containing the hopping operators, the types of commutativities and the
+            excitation indices.
+        """
+
+        if isinstance(self.num_modals, int):
+            num_modals = [self.num_modals] * self._molecule_data_transformed.num_modes
+        else:
+            num_modals = self.num_modals
+
+        return build_hopping_operators(num_modals, qubit_converter, excitations)
+
+    def interpret(self, raw_result: Union[EigenstateResult, EigensolverResult,
```

Comment:
@dlasecki If you extract `interpret` out of `MolecularProblem`, please do the same here